public/app/features/alerting/unified/rule-list/DataSourceGroupLoader.tsx

```diff
@@ -194,43 +194,53 @@ export function RulerBasedGroupRules({
       {pageItems.map((rulerRule) => {
         const promRule = matches.get(rulerRule);
 
-        return promRule ? (
-          <DataSourceRuleListItem
-            key={hashRule(promRule)}
-            rule={promRule}
-            rulerRule={rulerRule}
-            groupIdentifier={groupIdentifier}
+        if (promRule) {
+          // If rules are indistinguishable by name, labels, annotations, and query, we need to use the index to disambiguate
+          const promRuleIndex = promGroup.rules.indexOf(promRule);
```

Comment:
What would happen if we just used the `index` from `pageItems`? Wouldn't that guarantee a unique key for each list item too?